node/src/components/small_network/message.rs

```diff
@@ -260,15 +264,22 @@ impl<P: Display> Display for Message<P> {
                 protocol_version,
                 consensus_certificate,
                 is_joiner,
+                chainspec_hash,
             } => {
                 write!(
                     f,
-                    "handshake: {}, public addr: {}, protocol_version: {}, consensus_certificate: , is_joiner: {}",
```

Comment:
This was already broken, we need to fix it. Adding the `is_joiner: {}` at the end was wrong, the `write` only writes until the consensus certificate.

So, let's move the `is_joiner` to the left so that the string ends with `"consensus certificate: ".

Further down we need to add another `f.write_str(" chainspec_hash")`

Reply:
That's the quick fix. My preferred fix would be this though:

```rust
struct OptDisplay<'a, T>(&T);

impl<'a, T> Display for OptDisplay<'a, T> {
    #[inline]
    fn fmt(&self, f: Formatter<'_>) -> fmt::Result {
        match self {
            None => f.write_str("-")
            Some(val) => val.fmt(f)
        }
    }
}
```

and then use 

```rust
write!(f, "handshake: {}, public addr: {}, protocol_version: {}, consensus_certificate: {}, is_joiner: {}, chainspec_hash: {}",
network_name, public_addr, protocol_version, OptDisplay(&consensus_certificate) is_joiner, OptDisplay(&chainspec_hash)
```

Reply:
I like the `OptDisplay` way, but I'd go for writing "none" rather than "-" to be a little more explicit.  Not a big deal though.

Reply:
Done.